src/Compilers/Core/Portable/Diagnostic/DiagnosticSeverity.cs

```diff
@@ -27,7 +27,12 @@ public enum DiagnosticSeverity
         /// <summary>
         /// Something not allowed by the rules of the language or other authority.
         /// </summary>
-        Error = 3
+        Error = 3,
+
+        /// <summary>
+        /// Something that can be potentially improved.
+        /// </summary>
+        Suggestion = 4,
```

Comment:
> If we already have a couple of severities that the command line compile doesn't care about, why does this concern you?

Well, mistakes in past doesn't justify repeating them, especially if we know we could have done it better :)

> The compiler API is 100% built in support of IDE experiences, so I'm not sure I understand the push back.

Push back is not just specific to compiler having code that only IDE cares about. It is more about _Do these *truly* represent different severities?_ It doesn't seem so, because neither of them break my build.

Additionally, we are now confusing the diagnostic analyzer authors by exploding the enum. As analyzer author, when should I report an info versus a suggestion? Why should I have to care about this if all I want to do is report an issue that doesn't break end user's build?
Will we keep adding more of these fields in future as and when we feel we need different visual indicators are required in the IDE? If "Suggestion" gets the vote for being a new severity, why not "Unnecessary"? 

> I find a design where Suggestion is a custom tag for Info to be enormously fragile. With that, we would unnecessarily create invalid combinations of severities and tags since "Suggestion" is nonsensical when combined with "Error" and "Warning". If Suggestion is upgraded to a new severity, the combinations of tags and severities are all sensible.

I don't agree with this. Firstly, custom tags are associated with descriptors, i.e. rule metadata, not individual diagnostic instances. Rule metadata represents default information about category of issues created through the descriptor, as deemed appropriate by the analyzer author. Even the unnecessary custom tag added to faded diagnostic becomes non-sensical if user promotes the diagnostic to an error. We need to understand that custom tags is just metadata for class of diagnostics reported via a descriptor.